src/Sentry/Internal/Http/CachingTransport.cs

```diff
@@ -278,6 +286,8 @@ private async Task ProcessCacheAsync(CancellationToken cancellation)
                 await InnerProcessCacheAsync(file, cancellation).ConfigureAwait(false);
             }
 
+            SendFilesQueuedWhileOffline();
```

Comment:
Is this not already covered by the backoff function of your `NetworkConnectivityMonitor`?

Reply:
It is but that backoff function is eventually only checking every 32 seconds (and it could be configured to check even less frequently than that if we wanted).

This provides an alternative mechanism that detects network recovery immediately after any envelope is successfully enqueued and sent.